### ISSUE TITLE
SVG hyperlink support

### ISSUE THREAD

**Issue by bingenito** [badgen/badgen#57]
Query param options to specify an overall link for the entire badge or split links for left and right.  For dynamically generated tags, this will make usage from an object or embed element easier without having to further wrap img usage of the svg with another link as the link can be generated for you.

**Comment by bingenito** [comment]
I can PR this but wanted to open for discussion first before surprise!

**Comment by amio** [comment]
😅 Not sure if I get the correct idea, would you give an example usage of this?

**Comment by bingenito** [comment]
@amio Using embed and object tags to embed svg within a page allows for more advanced svg.

```
<embed src="<url to svg>" />
```

This includes providing an icon via a url and not svg, or adding support for hyperlinks within the badge itself.   A use case can be something like the github badges where it automatically embeds in the svg a link to the github project.  Then when hosted within a page using embed or object, the link is auto generated and included rather than having to wrap the image itself in a link.

With that said, while playing around quite a bit the actual scenarios to be able to leverage this are so narrow that perhaps it isn't worth the effort.  Most markdown sanitizes object and embed away so it leaves only manual html usage as a target.  I'm starting to think the this small use case isn't worth the effort.

**Comment by amio** [comment]
Agreed, sanitizers would be an issue to this in real world use cases.

While supporting `link` / `labelLink` / `statusLink` might be a meaningful feature for `badgen` library, if it can be achieved at a small cost.

**Comment by evelynhathaway** [comment]
I'm going to chime with some accessibility tips for when this gets implemented since I just implemented some a11y changes to shields, which has link support.

- The inner text of the link should be valid
- The link should not be inside or have `aria-hidden="true"`
- The `role` for the SVG should be the default or `group`
- You may run into some issues with IE or NVDA

See: https://github.com/badges/shields/pull/5289
